### PR TITLE
model settings None=not defined

### PR DESCRIPTION
**Pull request by memsharded**
Close https://github.com/conan-io/conan/issues/11176

### CODE REVIEW

**Review by czoido — APPROVED** [review]

File: conans/test/integration/settings/built_type_setting_test.py [hunk 1]
```diff
@@ -48,15 +48,15 @@ def test(self):
...
 
-        # This won't fail, as it has a build_type=None, which is allowed
         client.run("export . --name=pkg --version=0.1 --user=lasote --channel=testing")
-        client.run("install --requires=pkg/0.1@lasote/testing -pr=myprofile --build='*'")
-        self.assertEqual(1, str(client.out).count("BUILD TYPE: Not defined"))
+        client.run("install --requires=pkg/0.1@lasote/testing -pr=myprofile --build='*'",
+                   assert_error=True)
+        assert "ERROR: Invalid setting 'None' is not a valid 'settings.build_type'" in client.out
+        assert "Possible values are [None, 'Debug', 'Release', 'RelWithDebInfo', 'MinSizeRel']" in \
+            client.out
```
Comment (franramirez688) [inline conans/test/integration/settings/built_type_setting_test.py:61]: A bit confusing... We're saying that `None` isn't a valid one but it's also a possible one:
```bash
ERROR: Invalid setting 'None' is not a valid 'settings.build_type'
Possible values are [None, 'Debug', 'Release', 'RelWithDebInfo', 'MinSizeRel']
```
I think we should say instead:
```
ERROR: Invalid setting 'None' is not a valid 'settings.build_type'
Possible values are ['Debug', 'Release', 'RelWithDebInfo', 'MinSizeRel']
```
I guess that `null` is enough documented to know that it means you can keep empty that field

**Review by lasote — APPROVED** [review]

File: conans/test/integration/settings/built_type_setting_test.py (hunk 1 above)
Comment (memsharded) [reply conans/test/integration/settings/built_type_setting_test.py:61]: Good point, I'll filter it in the output

Comment (memsharded) [reply conans/test/integration/settings/built_type_setting_test.py:61]: Done: https://github.com/conan-io/conan/pull/11182/commits/9ccce9f4af3820160c337df53c745973dbc480db

**Review by lasote — APPROVED** [review]

**Review by franramirez688 — APPROVED** [review]

**Comment by System-Arch** [comment]
I believe this change introduced a bug that I am seeing while trying to use Conan 2.0.0-beta3. The issue relates to the following change in settings.py: validate()

ORIGINAL: 
  def validate(self):
        if self._value is None and "None" not in self._definition:

WITH THIS CHANGE:
    def validate(self):
        if self._value is None and None not in self._definition:

The error that I am seeing:
ERROR: conanfile.py (gmp/6.2.1): Invalid ID: Invalid: 'settings.compiler.threads' value not defined

Stepping through the code in the debugger reveals:
(Pdb) p self._value
None
(Pdb) p self._name
'settings.compiler.threads'
(Pdb) p self._definition
['None', 'posix', 'win32', 'ANY']

Thus, it seems the quotes should be restored around 'None' in the "if" statement cited above or the code that reads in the settings.yml file needs to treat None as a Python value rather than the string it appears to be read in as.

Thanks

**Comment by memsharded** [comment]
Hi @System-Arch 

Could you please report this as a new issue? Better than the code internals, it would be much better if you provide a simple but full ``conanfile.py`` that we could reproduce the behavior you are describing. Thanks!

**Pull request merged**